kustomize/internal/commands/build/flagaddmanagedby.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package build
+
+import (
+	"github.com/spf13/pflag"
+
+	"sigs.k8s.io/kustomize/api/konfig"
+)
+
+const (
+	flagEnableManagedbyLabelName = "enable_managedby_label"
+	flagEnableManagedbyLabelHelp = `enable adding ` + konfig.ManagedbyLabelKey
+)
+
+var (
+	flagEnableManagedbyLabelValue = true
```

Comment:
If the flag is turned on by default, we'll have to bump the major version number as tests that aren't part of this repo will break. It's backward incompatible behavior.
  
The feature's purpose is to help understand kustomize usage, not change change configuration.  So we're allowing a flag to modify how `build` works, as an exception to our [rules](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/eschewedFeatures.md#build-time-side-effects-from-cli-args-or-env-variables).  Since we allow a flag, we can allow an env var too.

So, if we also honored an env var, e.g. `KUSTOMIZE_ENABLE_MANAGEDBY_LABEL`, then we could turn this on/off in processes without demanding a flag, e.g. we could turn it on/off for all e2e tests by adding one line in [testExamplesE2EAgainstKustomize.sh](https://github.com/kubernetes-sigs/kustomize/blob/master/hack/testExamplesE2EAgainstKustomize.sh).  Likewise in other environments.

But still, we have to make the default false for now.

Reply:
Updated this change by
- set the flag default to false
- add the support for reading the env var
- To decide if the managedby label should be added, first read the flag value, then read the env value.